reflex/state.py

```diff
@@ -820,13 +851,30 @@ async def _process_event(
             payload: The event payload.
 
         Yields:
-            Tuple containing:
-                0: The state update after processing the event.
-                1: Whether the event is the final event.
+            StateUpdate object
         """
         # Get the function to process the event.
         fn = functools.partial(handler.fn, state)
 
+        token = self.get_token()
+
+        def as_state_update(events, final) -> StateUpdate:
```

Comment:
Should we lift this outside of the function, so that it's not redefined every time we call `_process_event`?